packages/xo-web/src/common/form/number.js

```diff
@@ -7,46 +7,54 @@ import decorate from '../apply-decorators'
 // it provide `data-*` to add params to the `onChange`
 const Number_ = decorate([
   provideState({
+    initialState: ({ value }) => {
+      if (value == null) {
+        value = ''
+      }
+      return {
+        value: String(value),
+      }
+    },
     effects: {
-      onChange: (_, { target: { value } }) => (state, props) => {
+      onChange(_, { target: { value } }) {
+        const { state, props } = this
         value = value.trim()
-        if (value === '') {
-          value = undefined
-        } else {
-          value = +value
-          if (Number.isNaN(value)) {
-            return
-          }
-        }
 
-        const params = {}
+        let params = {}
         let empty = true
         Object.keys(props).forEach(key => {
           if (key.startsWith('data-')) {
             empty = false
             params[key.slice(5)] = props[key]
           }
         })
+        params = empty ? undefined : params
+
+        if (value === '') {
+          props.onChange(undefined, params)
+        } else if (!Number.isNaN(+value) && +value !== +state.value) {
+          props.onChange(+value, params)
```

Comment:
The `type` of `input` is a `number`, why do you cast it here ?

Reply:
The `input` of type `number` emit `string` values.